interfaces/evse_board_support.yaml

```diff
@@ -0,0 +1,123 @@
+description: >-
+  This interface defines the board support driver for AC or DC minimal power path: ControlPilot,
+  output contactors. 
+  Other components of the power path such as IMD(DC)/RCD(AC)/Connector Lock etc have their own interfaces.
+cmds:
+  setup:
+    description: Setup config options
+    arguments:
+      three_phases:
+        description: 'true: Three phases enabled, false: only single phase'
+        type: boolean
+      has_ventilation:
+        description: 'true: Allow mode D charging, false: do not allow mode D charging'
+        type: boolean
+      country_code:
+        description: A two-letter country code in ISO 3166-1 alpha-2 format
+        type: string
+  get_hw_capabilities:
+    description: >-
+      Get Hardware capability/limits. For AC these are the limits of the power path (e.g. relais etc). 
+      For DC, these are the limits for the AC input of the ACDC converter stack, i.e. the complete AC input.
+      Note that DC output limits are reported by the DC power supply itself.
+    result:
+      description: Hardware capability/limits
+      type: object
+      $ref: /evse_board_support#/HardwareCapabilities
+  enable:
+    description: >-
+      Enables or disables the charging port. Typically disabled results in control
+      pilot state F. It must not accept cars for new charging sessions if disabled.
+    arguments:
+      value:
+        description: 'True: enabled, false: disabled.'
+        type: boolean
+  pwm_on:
+    description: Turns PWM on with duty cycle (in percent)
+    arguments:
+      value:
+        description: PWM duty cycle (>0, <100)
+        type: number
+        minimum: 0
+        maximum: 100
+  pwm_off:
+    description: Turns PWM off (constant high voltage)
+  pwm_F:
+    description: Turns PWM off with Error F (constant negative voltage)
+  allow_power_on:
+    description: >-
+      Sets allow_power_on flag. If false, Relais must never be switched
+      on.
+    arguments:
+      value:
+        description: Flag and context
+        type: object
+        $ref: /evse_board_support#/PowerOnOff
+  ac_switch_three_phases_while_charging:
+    description: >-
+      Special command to force switching between one and three phases while
+      charging is active. HW must go through some special sequence to ensure safe
```

Comment:
Is it really *must* and IMHO its rather *a* special sequence or some special sequences

Reply:
IEC61851-1 does not support switching phases during charging. It is a non-standard feature that needs to be explicilty supported by the BSP driver and the BSP will need to implement a special sequence to do the switching. Many use state F for that but there are other implementations as well. Note that some cars will be destroyed permanently if you do not take special care here in the BSP.

Reply:
So I would not say that this is a must for any BSP to support, most BSP should not support this. For AC basic charging applications with solar where this is useful it can be implemented in the BSP and the BSP needs to then advertise it in its capabilities

Reply:
Looking into the future this should be entirely replaced with ISO-2 supporting 3ph/2A charging. But it will take some years until car manufacturers implement that if at all.

Reply:
Do you have a suggestion on how to rewrite the description to make it more clear?

Reply:
would it make sense to move this to a different interface then? So its clearer what is an optional interface and what not? This would also help to identify the interfaces which are rather for debugging and not for production

Reply:
This could also be true for `evse_replug`

Reply:
Clarfied the description